### PR TITLE
Refactor the Matrix class

### PR DESCRIPTION
**Pull request by mikestone14**

### CODE REVIEW

File: views/components/Matrix.d3.js [hunk 1]
```diff
@@ -1,454 +1,599 @@
...
-import { stringToDate } from '../../server/utils/dateConverter';
-import basePathConfig from '../../server/configs/basePathConfig';
+import * as d3 from 'd3'
 
-const basePath = basePathConfig || '';
+import { getColor } from '../../server/utils/invertColor'
+import { stringToDate } from '../../server/utils/dateConverter'
+import basePathConfig from '../../server/configs/basePathConfig'
+
+const basePath = basePathConfig || ''
```
Comment (taylorkearns) [inline views/components/Matrix.d3.js:7]: It looks like basePathConfig already [returns an empty string](https://github.com/AMP-SCZ/dpdash/blob/pnl-devel/server/configs/basePathConfig.js)?

File: views/components/Matrix.d3.js [hunk 2]
```diff
@@ -1,454 +1,599 @@
...
+  constructor(el, props) {
+    this.el = el
+    this.cardSize = props.cardSize
+    this.halfCardSize = props.cardSize / 2
+    this.startDay = props.startDay
+    this.lastDay = props.lastDay
+    this.maxDay = props.maxDay
+    this.initialHeight = props.height
+    this.height = props.height - margin.top - margin.bottom
+    this.width = props.width - margin.right
```
Comment (taylorkearns) [inline views/components/Matrix.d3.js:32]: Do we need to account for left margin?

Comment (mikestone14) [reply views/components/Matrix.d3.js:32]: I'm not sure. This should be a pure refactor without any functional changes so this value should be what the existing `width` value is.

File: views/components/Matrix.d3.js (hunk 1 above)
Comment (mikestone14) [reply views/components/Matrix.d3.js:7]: yeah good call. There's still a ton of refactor opportunity in this class. Trying not to get too carried away but this is a change I can make.

**Comment by mikestone14** [comment]
closing in favor of #328

**Pull request closed without merging**